src/data/mystery-encounters/mystery-encounter.ts

```diff
@@ -0,0 +1,963 @@
+  excludePrimaryFromSupportRequirements: boolean;
+
+  dialogue: MysteryEncounterDialogue;
+  enemyPartyConfigs: EnemyPartyConfig[];
+
+  dialogueTokens: Record<string, string>;
+  expMultiplier: number;
+}
+
+/**
+ * MysteryEncounter class that defines the logic for a single encounter
+ * These objects will be saved as part of session data any time the player is on a floor with an encounter
+ * Unless you know what you're doing, you should use MysteryEncounterBuilder to create an instance for this class
+ */
+export default class MysteryEncounter implements IMysteryEncounter {
+  /**
+   * Required params
+   */
+  encounterType: MysteryEncounterType;
+  options: [MysteryEncounterOption, MysteryEncounterOption, ...MysteryEncounterOption[]];
+  spriteConfigs: MysteryEncounterSpriteConfig[];
+  /**
+   * Optional params
+   */
+  encounterTier: MysteryEncounterTier;
+  /**
+   * Custom battle animations that are configured for encounter effects and visuals
+   * Specify here so that assets are loaded on initialization of encounter
+   */
+  encounterAnimations?: EncounterAnim[];
+  /**
+   * If specified, defines any game modes where the MysteryEncounter should *NOT* spawn
+   */
+  disabledGameModes?: GameModes[];
+  /**
+   * If true, hides "A Wild X Appeared" etc. messages
+   * Default true
+   */
+  hideBattleIntroMessage: boolean;
+  /**
+   * If true, when an option is selected the field visuals will fade out automatically
+   * Default false
+   */
+  autoHideIntroVisuals: boolean;
+  /**
+   * Intro visuals on the field will slide in from the right instead of the left
+   * Default false
+   */
+  enterIntroVisualsFromRight: boolean;
+  /**
+   * If true, allows catching a wild pokemon during the encounter
+   * Default false
+   */
+  catchAllowed: boolean;
+  /**
+   * If true, encounter will continuously run through multiple battles/puzzles/etc. instead of going to next wave
+   * MUST EVENTUALLY BE DISABLED TO CONTINUE TO NEXT WAVE
+   * Default false
+   */
+  continuousEncounter: boolean;
+  /**
+   * Maximum number of times the encounter can be seen per run
+   * Rogue tier encounters default to 1, others default to 3
+   */
+  maxAllowedEncounters: number;
+  /**
+   * If true, encounter will not animate the target Pokemon as part of battle animations
+   * Used for encounters where it is not a "real" battle, but still uses battle animations and commands (see {@linkcode FunAndGamesEncounter} for an example)
+   */
+  hasBattleAnimationsWithoutTargets: boolean;
+  /**
+   * If true, will skip enemy pokemon turns during battle for the encounter
+   * Used for encounters where it is not a "real" battle, but still uses battle animations and commands (see {@linkcode FunAndGamesEncounter} for an example)
+   */
+  skipEnemyBattleTurns: boolean;
+  /**
+   * If true, will skip COMMAND input and go straight to FIGHT (move select) input menu
+   */
+  skipToFightInput: boolean;
+
+  /**
+   * Event callback functions
+   */
```

Comment:
you could use 
```suggestion
  // #region Event callback functions
```

Make sure to close it with
```ts
  // #endregion
```

This would show up like this on vs-codium minimap:
<img width="166" alt="image" src="https://github.com/user-attachments/assets/a4e31814-77c8-4a47-a112-2fcdc11afe46">